test/WebJobs.Script.Tests/Workers/Rpc/GrpcWorkerChannelTests.cs

```diff
@@ -145,6 +145,48 @@ public async Task StartWorkerProcessAsync_ThrowsTaskCanceledException_IfDisposed
             });
         }
 
+        [Fact]
+        public void WorkerChannel_Dispose_With_WorkerTerminateCapability()
+        {
+            var initTask = _workerChannel.StartWorkerProcessAsync(CancellationToken.None);
+
+            IDictionary<string, string> capabilities = new Dictionary<string, string>()
+            {
+                { RpcWorkerConstants.HandlesWorkerTerminateMessage, "1" }
+            };
+
+            StartStream startStream = new StartStream()
+            {
+                WorkerId = _workerId
+            };
+
+            StreamingMessage startStreamMessage = new StreamingMessage()
+            {
+                StartStream = startStream
+            };
+
+            // Send worker init request and enable the capabilities
+            GrpcEvent rpcEvent = new GrpcEvent(_workerId, startStreamMessage);
+            _workerChannel.SendWorkerInitRequest(rpcEvent);
+            _testFunctionRpcService.PublishWorkerInitResponseEvent(capabilities);
+
+            _workerChannel.Dispose();
+            var traces = _logger.GetLogMessages();
+            var expectedLogMsg = $"Sending WorkerTerminate message with grace period {WorkerConstants.ProcessExitTimeoutInMilliseconds}.";
+            Assert.True(traces.Any(m => string.Equals(m.FormattedMessage, expectedLogMsg)));
```

Comment:
Is it possible for us to add an E2E test where we check if we actually sent a WorkerTerminate message, and assert the value of `GracePeriod`?

Reply:
Wonder there is a way for us to use a different wait time for the test runs so that UTs are done faster.

Reply:
Yes, that's a good point. On checking code, it seems we will need some changes in the test setup so that we can have a `TestConstants` file which will be passed in the classes. On quick check, found another test which waits for actual time (25 sec) instead of shorter one. https://github.com/Azure/azure-functions-host/blob/5194284aaeecf1734a14e45871930d9cc2c7f813/test/WebJobs.Script.Tests/HttpWorker/DefaultHttpWorkerServiceTests.cs#L481

There is an old issue #8366 , I created for testing timeouts. May be, we can also add this scenario there to cover improvements in tests for time related scenarios.